### PR TITLE
Reminders socket

### LINKED ISSUE #7: Notifications
Users can get notified (also see command notifcation settings) by things like handing in assignments and more.

### PR DESCRIPTION
**Pull request by PandaGamersHDNL**

### CODE REVIEW

File: src/models/reminder.ts [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
+  target: {
+    user: string
+  }
 }
+
+export interface AssignmentDM {
+  id: string
+  assignmentID:string
+  userDiscordID: string,
+  message: MessageEmbedOptions
```
Comment (DrunkenToast) [inline src/models/reminder.ts]: nitpick: rename to embed for consistency

File: src/services/shard-service.ts [hunk 2]
```diff
@@ -50,19 +51,31 @@ export class ShardService {
...
+
+    this.socket.on('reminderGuild', (data: GuildReminder) => {
+      if (this.client !== undefined)
+        ReminderService.sendGuildReminder(data, this.client);
+    });
+
+    this.socket.on('reminderGuild', (data: UserReminder) => {
+      if (this.client !== undefined)
+        ReminderService.sendUserReminder(data, this.client);
+    });
```
Comment (DrunkenToast) [inline src/services/shard-service.ts]: why twice?

**Review by DrunkenToast — APPROVED** [review]

File: src/services/shard-service.ts (hunk 2 above)
Comment (PandaGamersHDNL) [reply src/services/shard-service.ts]: copy paste error

**Review by xunafay — APPROVED** [review]

File: src/models/reminder.ts (hunk 1 above)
Comment (PandaGamersHDNL) [reply src/models/reminder.ts]: message = string/ msg embed

**Pull request merged**